### PR TITLE
Rename RoadObjectsStore to RoadObjectStore

### LINKED ISSUE #3041: Rename RoadObjectsStore to RoadObjectStore
We should rename RoadObjectsStore and RoadObjectsStoreDelegate to RoadObjectStore and RoadObjectStoreDelegate, respectively. The latter names are more grammatical and consistent with the majority of types that begin with “RoadObject”.

/cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by ShanMa1991**
### Description
This pr is to fix #3041 to rename `RoadObjectsStore` and `RoadObjectsStoreDelegate` to `RoadObjectStore` and `RoadObjectStoreDelegate`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -57,6 +57,7 @@
 * Added `RoadObjectsStore.addUserDefinedRoadObject(_:)`, `RoadObjectsStore.removeUserDefinedRoadObject(identifier:)`, and `RoadObjectsStore.removeAllUserDefinedRoadObjects()` for adding/removing user-defined road objects to the electronic horizon. ([#3004](https://github.com/mapbox/mapbox-navigation-ios/pull/3004))
 * Removed `Alert` enum, and `alert`, `distance`, `length`, `beginCoordinate`, `endCoordinate`, `beginSegmentIndex`, and `endSegmentIndex` properties from `RouteAlerts`. ([#2991](https://github.com/mapbox/mapbox-navigation-ios/pull/2991))
 * Added the `RouteAlerts.roadObject` property. ([#2991](https://github.com/mapbox/mapbox-navigation-ios/pull/2991))
+* Renamed `RoadObjectsStore` and `RoadObjectsStoreDelegate` to `RoadObjectStore` and `RoadObjectStoreDelegate` respectively. ([#3045](https://github.com/mapbox/mapbox-navigation-ios/pull/3045))
```
Comment (1ec5) [inline CHANGELOG.md:60]: RoadObjectsStore and RoadObjectsStoreDelegate were first introduced in a previous v2.0.0 beta, so we would mention this rename in the next beta’s release notes but not this running changelog.

File: Sources/MapboxCoreNavigation/EHorizon/RoadObjectStore.swift [hunk 2]
```diff
@@ -37,7 +37,7 @@ public final class RoadObjectsStore {
 
     /**
      Returns road object with given identifier, if such object cannot be found returns null.
-     NB: since road objects can be removed/added in background we should always check return value for null,
+     NB: since road object can be removed/added in background we should always check return value for null,
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/EHorizon/RoadObjectStore.swift:40]: Nit: It’s more grammatical to say either “road objects” or “a road object” when referring to an instance of the RoadObject class in a sentence. (×5)

**Review by 1ec5 — APPROVED** [review]
There’s some overcorrection in the documentation comments, but otherwise the refactor appears to have gone smoothly.

File: CHANGELOG.md (hunk 1 above)
Comment (ShanMa1991) [reply CHANGELOG.md:60]: I see. I'll remove this one. And should we also update the name of the `RoadObjectsStore` and `RoadObjectsStoreDelegate` to `RoadObjectsStore` and `RoadObjectsStoreDelegate` in this changeLog list?

File: Sources/MapboxCoreNavigation/EHorizon/RoadObjectStore.swift (hunk 2 above)
Comment (ShanMa1991) [reply Sources/MapboxCoreNavigation/EHorizon/RoadObjectStore.swift:40]: Thanks for the suggestion, I'll improve this part.

**Pull request merged**